src/Streamly/Internal/Control/Concurrent.hs

```diff
@@ -31,9 +46,14 @@ import Control.Monad.Trans.Control (MonadBaseControl, control, StM)
 -- /Since: 0.1.0 ("Streamly")/
 --
 -- @since 0.8.0
-type MonadAsync m = (MonadIO m, MonadBaseControl IO m, MonadThrow m)
+type MonadAsync m = (MonadRunInIO m, MonadThrow m)
```

Comment:
I am not sure if this will have user visible API impact. Especially with type applications, and this type may come up in type errors as well. To be safe, for now, we can put this under ifdef USE_UNLIFTIO and expand this to (MonadIO m, MonadBaseControl IO m, MonadThrow m) in the default case.